src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1701,16 +1707,55 @@ private IEnumerable<PSTypeName> InferTypesFrom(MemberExpressionAst memberExpress
 
         private static void AddTypesFromMethodCacheEntry(
             DotNetAdapter.MethodCacheEntry methodCacheEntry,
+            IList<ITypeName> genericTypeArguments,
             List<PSTypeName> result,
             bool isInvokeMemberExpressionAst)
         {
             if (isInvokeMemberExpressionAst)
             {
+                Type[] resolvedTypeArguments = null;
+                if (genericTypeArguments is not null)
+                {
+                    resolvedTypeArguments = new Type[genericTypeArguments.Count];
+                    for (int i = 0; i < genericTypeArguments.Count; i++)
+                    {
+                        Type resolvedType = genericTypeArguments[i].GetReflectionType();
+                        if (resolvedType is null)
+                        {
+                            return;
+                        }
```

Comment:
We probably shouldn't return here. There are cases where a generic argument cannot be resolved until at runtime (e.g. an assembly gets loaded at runtime that contains the needed type), but the return type for the method may not contain a generic argument. In such cases, we can get the type needed for infer without a problem.

Here is an example:
```
PS:56> $s = [System.Collections.Concurrent.ConcurrentDictionary[string, string]]::new()
PS:57> $s.AddOrUpdate

OverloadDefinitions
-------------------
string AddOrUpdate[TArg](string key, System.Func[string,TArg,string] addValueFactory, System.Func[string,string,TArg,string] updateValueFactory,
TArg factoryArgument)
string AddOrUpdate(string key, System.Func[string,string] addValueFactory, System.Func[string,string,string] updateValueFactory)
string AddOrUpdate(string key, string addValue, System.Func[string,string,string] updateValueFactory)
```

Reply:
Please be noted that I pushed a commit, so please sync first when you make further changes.

Reply:
Good point, I've removed the early return.